src/ui/components/componentmanager.js

```diff
@@ -152,6 +154,16 @@ export default class ComponentManager {
         return component;
       }
 
+      this._core.storage.registerListener({
+        eventType: 'update',
+        storageKey: component.moduleId,
+        callback: (data) => {
+          component.setState(data);
```

Comment:
Are we going to have two, duplicate `setState` calls on certain updates? The duplicate calls would happen only for those `StorageKeys` that have been cut-over I think. Once the full cut-over is complete, the we'd just have the one listener per key?

Reply:
That's a valid concern! The plan is to fully cut over each storage key. As long as a storage key like NAVIGATION is fully cut over in a single PR, there's no chance for duplicate updates, since all older setState() calls will be updated to the new set() calls

Reply:
Got it! Sounds good!